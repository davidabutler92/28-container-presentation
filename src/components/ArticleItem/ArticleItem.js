import React from 'react';
import PropTypes from 'prop-types';

function ArticleItem({ title, author, description }) {
  return (
    <>
      <h1>{title}</h1>
      <p>{author}</p>
      <p>{description}</p>
    </>
  );
}

ArticleItem.propTypes = {
  title: PropTypes.string,
  author: PropTypes.string,
  description: PropTypes.string,
};

export default ArticleItem;
