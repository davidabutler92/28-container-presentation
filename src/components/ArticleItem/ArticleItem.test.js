import React from 'react';
import { render, cleanup } from '@testing-library/react';
import ArticleItem from './ArticleItem';

describe('ArticelItem component', () => {
  afterEach(() => cleanup());
  it('renders a single article', () => {
    const { asFragment } = render(
      <ArticleItem 
        title="hello" 
        author="some author"
        description="something else described.."
      />);
    expect(asFragment()).toMatchSnapshot();
  });
});
