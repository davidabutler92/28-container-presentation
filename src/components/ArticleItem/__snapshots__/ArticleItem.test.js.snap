// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticelItem component renders a single article 1`] = `
<DocumentFragment>
  <h1>
    hello
  </h1>
  <p>
    some author
  </p>
  <p>
    something else described..
  </p>
</DocumentFragment>
`;
